server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -0,0 +1,392 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+import reactor.core.publisher.Flux;
+
+class DeletedMessagesVaultRoutesTest {
+
+    private WebAdminServer webAdminServer;
+    private MemoryDeletedMessagesVault vault;
+    private InMemoryMailboxManager mailboxManager;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        vault = spy(new MemoryDeletedMessagesVault());
+        InMemoryIntegrationResources inMemoryIntegrationResources = new InMemoryIntegrationResources();
+        InMemoryIntegrationResources.Resources inMemoryResource = inMemoryIntegrationResources.createResources(new SimpleGroupMembershipResolver());
+        mailboxManager = spy(inMemoryResource.getMailboxManager());
+
+        MemoryTaskManager taskManager = new MemoryTaskManager();
+        JsonTransformer jsonTransformer = new JsonTransformer();
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new TasksRoutes(taskManager, jsonTransformer),
+            new DeletedMessagesVaultRoutes(vault, mailboxManager, jsonTransformer, taskManager));
+
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(DeletedMessagesVaultRoutes.ROOT_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void afterEach() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void restoreShouldReturnInvalidWhenUserIsInvalid() {
+        when()
+            .post("not@valid@user.com")
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .body("statusCode", is(400))
+            .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+            .body("message", is("The username should not contain multiple domain delimiter."));
+    }
+
+    @Test
+    void postShouldReturnNotFoundWhenNoUserPathParameter() {
+        when()
+            .post()
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("statusCode", is(404))
+            .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+            .body("message", is("POST /deletedMessages/user can not be found"));
+    }
+
+    @Test
+    void restoreShouldReturnATaskCreated() {
+        when()
+            .post(USER.asString())
+        .then()
+            .statusCode(HttpStatus.CREATED_201)
+            .body("taskId", notNullValue());
+    }
+
+    @Test
+    void restoreShouldProduceAnSuccessfulTaskWithAdditionalInformation() {
+        vault.append(USER, DELETED_MESSAGE).block();
+        vault.append(USER, DELETED_MESSAGE_2).block();
+
+        String taskId = given()
+            .when()
```

Comment:
Why not `with()` directly ?

Reply:
because I don't know this method exist :)